Assets/Scripts/Environment/Hangar/Battery/BatteryInteractable.cs

```diff
@@ -24,7 +24,9 @@ public void TakeBattery()
 
         batteryModelCollider.enabled = false;
 
-        ConfigureRigidbody(isConnectingToPlayer: true);
+        ConfigureRigidbody(isConnectingToPlayer: true);
+
+        HangarManager.currentBatteryBeingHeld = gameObject.GetComponent<BatteryWrapper>();
```

Comment:
You can just call ``GetComponent<BatteryWrapper>();`` here. Don't actually need the ``gameObject``.